### PR TITLE
adding tabswitcher component

### LINKED ISSUE #262: Build tabswitcher component
There's a tabswitcher used in a lot of the views, so let's build a component!

### PR DESCRIPTION
**Pull request by gvn**
Closes #262

### CODE REVIEW

**Comment by gvn** [comment]
I think mobile is going to need some additional design thought, so will probably do a follow up for that.

File: app/react/components/tab-switcher/tab-switcher.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React from "react";
+
+export default React.createClass({
+  propTypes: {
+    className: React.PropTypes.string
```
Comment (alanmoo) [inline app/react/components/tab-switcher/tab-switcher.jsx]: Is it possible to define the shape of props.children here?

File: app/react/components/tab-switcher/tab-switcher.scss [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      &:last-child {
+        margin-right: 0;
+      }
+
+      &.active {
+        background: $off-white;
+        color: inherit;
+      }
+
+      border: 0;
```
Comment (alanmoo) [inline app/react/components/tab-switcher/tab-switcher.scss]: Not sure if we have this covered in style, but should styles targeting the element be above nested styles? As in, this block then `&:first-child{}`

File: app/react/components/tab-switcher/tab-switcher.scss [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    display: flex;
+
+    button {
+      &:first-child {
+        margin-left: 0;
+      }
+
+      &:last-child {
+        margin-right: 0;
+      }
```
Comment (alanmoo) [inline app/react/components/tab-switcher/tab-switcher.scss]: Minor nit: I like to do `& + button{margin-left: 0.5rem}` as it's a little less code and clearer at once. Maybe we should standardize on something in general.

File: app/react/components/tab-switcher/tab-switcher.scss [hunk 4]
```diff
@@ -0,0 +1,51 @@
+.tab-switcher {
+  .tabs {
+    display: flex;
+
+    button {
```
Comment (alanmoo) [inline app/react/components/tab-switcher/tab-switcher.scss]: Use a class instead?

File: app/react/components/tab-switcher/tab-switcher.scss [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      flex: 1;
+      margin: 0 0.5rem;
+      text-transform: none;
+      color: $white;
+      background: rgba($off-white, 0.2);
+    }
+  }
+
+  &.inline .tabs {
+    button {
```
Comment (alanmoo) [inline app/react/components/tab-switcher/tab-switcher.scss]: Use a class?

**Comment by alanmoo** [comment]
Two things beyond line comments:
1. Can you remove the "shift" of text when changing tabs? I think it's border related, maybe change the box-sizing on the tabs or make it `outline` instead of `border`.

2. We're going to change this to better support routing in the future, right? Seems like it has the potential to load a lot of markup all at once (though it's all in the bundle so maybe it's not as relevant)

File: app/react/components/tab-switcher/tab-switcher.jsx (hunk 1 above)
Comment (gvn) [inline app/react/components/tab-switcher/tab-switcher.jsx]: Like as an array of elements or what?

Comment (alanmoo) [inline app/react/components/tab-switcher/tab-switcher.jsx]: Yeah

Comment (alanmoo) [inline app/react/components/tab-switcher/tab-switcher.jsx]: Well, I guess that's always the case?

File: app/react/components/tab-switcher/tab-switcher.scss (hunk 3 above)
Comment (gvn) [inline app/react/components/tab-switcher/tab-switcher.scss]: I'm not sure how that's clearer, but it is less code I guess...

File: app/react/components/tab-switcher/tab-switcher.scss (hunk 2 above)
Comment (gvn) [inline app/react/components/tab-switcher/tab-switcher.scss]: Yeah, that makes sense.

File: app/react/components/tab-switcher/tab-switcher.scss (hunk 3 above)
Comment (alanmoo) [inline app/react/components/tab-switcher/tab-switcher.scss]: Clearer (in my mind, at least) as a single selector describing a common pattern

File: app/react/components/tab-switcher/tab-switcher.jsx (hunk 1 above)
Comment (gvn) [inline app/react/components/tab-switcher/tab-switcher.jsx]: Yeah, I mean I can add it...it doesn't hurt. Wouldn't give us much other than a warning if someone has a tabswitcher with under 2 children.

Comment (alanmoo) [inline app/react/components/tab-switcher/tab-switcher.jsx]: I was thinking more about if there's content that's not just a container, but really any child element would make it a container.  So forget I said anything 😅

Comment (gvn) [inline app/react/components/tab-switcher/tab-switcher.jsx]: Lol, no worries. 👍

**Comment by gvn** [comment]
1. Fixed
2. Probably, but that feels like R2 to me.

**Comment by alanmoo** [comment]
Was still seeing the shift. See https://github.com/gvn/science.mozilla.org/pull/2 for fix

**Comment by alanmoo** [comment]
If you're good merging that PR in to your branch, then this can be merged in.

**Pull request merged**